### ISSUE TITLE
`indent` rule wants to add trailing newlines/whitespaces to multi-line raw string literals

### ISSUE THREAD

**Issue by RBusarow** [pinterest/ktlint#1375]
## Expected Behavior
The `indent` rule should not change the values of raw strings.

## Observed Behavior
Since `0.44.0` (specifically from PR #1262), the `indent` rule changes the position of the closing quotes of any multi-line raw string literal so that they're aligned.  This isn't just a formatting change, since the added spaces and/or newline are added to the value of the actual string.

## Steps to Reproduce

Given this top-level (top level because the indent matters) property:
```kotlin
val someCodeBlock = """
  foo()"""
```
Formatting will change it to:
```kotlin
val someCodeBlock = """
  foo()
"""
```

## Your Environment 
* Version of ktlint used: `0.44.0`
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): Gradle plugin
* Link to your project (if it's a public repository): https://github.com/RBusarow/ModuleCheck

**Comment by romtsn** [comment]
I think that's what my concern was exactly about https://github.com/pinterest/ktlint/pull/1262#discussion_r739840536, but somehow it got ignored. We shouldn't change contents of raw strings if they are not followed by `trimIndent`/`trimMargin`

**Comment by RBusarow** [comment]
> We shouldn't change contents of raw strings if they are not followed by `trimIndent`/`trimMargin`

Not even then.

You can safely increase or decrease the indent/margin _of existing lines_ if one of those extensions is used, but adding a new line like this would still result in a change to the string's value.

**Comment by paul-dingemans** [comment]
> I think that's what my concern was exactly about [#1262 (comment)](https://github.com/pinterest/ktlint/pull/1262#discussion_r739840536), but somehow it got ignored. We shouldn't change contents of raw strings if they are not followed by `trimIndent`/`trimMargin`

I can now see what your concerns were. In this case it indeed is wrong to align the closing quotes when not followed by trim indent/trimmargin. As documented on https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/trim-indent.html this method (same for trimmargin) do remove the trailing and pending newlines. Of course this does not happen for a raw string without them.

**Comment by paul-dingemans** [comment]
@RBusarow Is your problem just an abstraction for a real problem in production code or just a semantic discussion about how you want to write unit tests? Semantically, you are still right, that this is unwanted behavior.
```
val someCodeBlock = """
  foo()"""
```

If above would be used in some production code like:
```
val message = """
  some-message"""
```
then this message would not not contain a newline at the beginning but also the indentation before the text `some-message`.

See unit tests below, which clearly identifies the behavior between the approaches:
```
@Test
fun `Given a multiline raw string without trimIndent contains a new line`() {
    val message = """
        some-message"""

    assertThat(message).isEqualTo("\n            some-message")
}

@Test
fun `Given a multiline raw string with trimIndent does not contain leading or trailing new lines`() {
    val message = """
        some-message
    """.trimIndent()

    assertThat(message).isEqualTo("some-message")
}
```

So is it really likely that this causes problems in production code?

**Comment by RBusarow** [comment]
@paul-dingemans That's just an abstraction.  My real-world example of 0.44.0's changes is here: https://github.com/RBusarow/ModuleCheck/pull/393/files.

This project parses Java, Kotlin, Groovy, and XML files, and performs auto-corrections in Groovy and Kotlin Gradle files.  It's important that whitespaces are handled correctly in tests, as well as in the changes being made to the Gradle files.  

The changes to this rule do not affect production code, but they affect my ability to test the production code.

**Comment by paul-dingemans** [comment]
@RBusarow @gchallen I am gonna fix this in the next release.

Meanwhile you can have a look at the IndentationRuleTest in ktlint. For example in https://github.com/pinterest/ktlint/blob/e1674f8810c8bed6d6ee43c7e97737855c848975/ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/IndentationRuleTest.kt#L1272 we use special markers when tabs, spaces or the raw string literal quotes are essential in the test. This way you can make those characters very explicit and easily readable.

**Comment by jrogers** [comment]
I saw this error in some code as well after updating. In this scenario the string literals (they were format templates) were defined as const and were relying on the specific whitespace formatting because the trimIndent() call couldn't be added without making them non-const.
